atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java

```diff
@@ -161,22 +161,26 @@ public Member getMember(final MemberId memberId) {
       this.localProperties.putAll(localMember.properties());
       discoveryService.addListener(discoveryEventListener);
 
-      LOGGER.info("{} - Member activated: {}", localMember.id(), localMember);
+      // we need to add our local node to the member list,
+      // to share the mapping between node id and address in the cluster
       localMember.setState(State.ALIVE);
       members.put(localMember.id(), localMember);
       post(new GroupMembershipEvent(GroupMembershipEvent.Type.MEMBER_ADDED, localMember));
 
+      LOGGER.debug("Nodes from discovery service {}", discoveryService.getNodes());
+
       registerHandlers();
       gossipFuture =
-          swimScheduler.scheduleAtFixedRate(
-              this::gossip, 0, config.getGossipInterval().toMillis(), TimeUnit.MILLISECONDS);
+          swimScheduler.schedule(
```

Comment:
These futures should be updated when they are scheduled again.

Reply:
true